### PR TITLE
subsys: download_client: Fix bug in header parser

### PR DESCRIPTION
**Pull request by jorgenmk**
When parsing http header, download_client checks whole http packet for header data, limit to checking http header area only.

### CODE REVIEW

**Review by simensrostad — APPROVED** [review]
lgtm

File: subsys/net/lib/download_client/src/http.c [hunk 1]
```diff
@@ -124,7 +124,7 @@ static int http_header_parse(struct download_client *client, size_t *hdr_len)
 {
 	char *p;
 	char *q;
-	unsigned int http_status;
+	unsigned long http_status;
```
Comment (eivindj-nordic) [inline subsys/net/lib/download_client/src/http.c:127]: When is this required to be `long`?

**Review by eivindj-nordic — APPROVED** [review]
One question, otherwise looks good to me!

File: subsys/net/lib/download_client/src/http.c (hunk 1 above)
Comment (jorgenmk) [reply subsys/net/lib/download_client/src/http.c:127]: It's only set in this line below:

**Pull request merged**